records_mover/records/schema/field/field_types.py

```diff
@@ -0,0 +1,16 @@
+from typing_extensions import Literal
+from typing_inspect import get_args
+from typing import List
+
+FieldType = Literal['integer',
+                    'decimal',
+                    'string',
+                    'boolean',
+                    'date',
+                    'time',
+                    'timetz',
+                    'datetime',
+                    'datetimetz']
+
+# Be sure to add new things below in FieldType, too
+RECORDS_FIELD_TYPES: List[str] = list(get_args(FieldType))  # type: ignore
```

Comment:
I renamed this from `types.py` as mypy doesn't support having a file named that when it's working in the mode it's used in for single-file IDE support.

I also dropped the guards, as we now include both typing_inspect and typing_extensions as dependencies.